core/variant.cpp

```diff
@@ -2687,7 +2689,7 @@ uint32_t Variant::hash() const {
 		} break;
 		case INT: {
 
-			return _data._int;
+			return _data._int >= 0 ? _data._int : UINT32_MAX + _data._int;
```

Comment:
I'm not sure this is a good idea.
While it was previously a bug, this custom handling now makes it a feature that several numbers have the same hash (and since `Variant::_data._int` is an `int64_t`, there should actually be a lot of them).

It's probably better to use `hash_one_uint64` instead, though it might break compatibility.

Reply:
I totally forgot that _int in this situation is 64 bit instead 32 bit, so this is wrong.